lib/traject/writers/solr_better_json_writer.rb

```diff
@@ -109,18 +114,76 @@ def max_sleep_seconds
     Float(2**@retry_count)
   end
 
-  def generate_json(batch)
-    arr = []
+  # Collection of Traject contexts to be sent to solr
+  class Batch
```

Comment:
I initially extracted it because I needed an underlying data structure that mapped desired actions to documents. That data structure could then be used to generate the JSON to solr as well as to report about the status of the actions. I could instead extract the reporting-related stuff to methods directly on the writer that take a `Batch`, I guess?

Reply:
Yeah, it feels to me like the reporting stuff should be separate if possible.

Reply:
Moved it to the `SdrEvents` class.